src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -3620,8 +3620,11 @@ private static bool IsValidSystemTimeZoneId(string id)
             Byte[] typeOfLocalTime, TZifType[] transitionTypes, Boolean[] StandardTime, Boolean[] GmtTime, string futureTransitionsPosixFormat)
         {
             // To generate AdjustmentRules, use the following approach:
-            // 1. The first AdjustmentRule goes from DateTime.Min to the first transition and uses the first standard transitionType
-            // (or the first transitionType if none of them are standard)
+            // 0. Skip any DateTime.MinValue transition times. In newer versions of the tzfile, there
+            // is a "big bang" transition time, which is before the year 0001. Since any times before year 0001
+            // cannot be represented by DateTime, there is no reason to make AdjustmentRules for these unrepresentable time periods.
+            // 1. If there are no DateTime.MinValue times, the first AdjustmentRule goes from DateTime.Min
```

Comment:
can we have a TZ starts at DateTime.MinValue and its UTC offset at that time is different than the current UTC offset? this can make the calculation wrong. 
what about you keep the rules for time zones starts at DateTime.MinValue and for big bang case we just skip. I mean having TZif_UnixTimeToDateTime return Boolean telling about success or fail and in calling site you just check the returned value and skip it if it is false.